### PR TITLE
Update first part of notebook 3 with more explanation of forwad models

### LINKED ISSUE #38: Update notebook 3

### PR DESCRIPTION
**Pull request by nabriis**
Closes #38 

- Replaced model with forward model in most cases.
- Added a bit more initial explanation of why forward models are important.
- Made section 4 not star (custom forward models)
- Made notation more consistent (used y^data and x^exact for specific vectors) and left x and y as random variables always.
- Minor typos etc.

### CODE REVIEW

**Comment by jakobsj** [comment]
Was this ready to be looked at now or were you making some extra changes to this this afternoon?

**Comment by nabriis** [comment]
> Was this ready to be looked at now or were you making some extra changes to this this afternoon?

Its ready. I quickly updated after stand-up with the comments.

File: training/Exercise03_Forward_Models.ipynb [hunk 1]
```diff
@@ -4,25 +4,32 @@
...
    "metadata": {},
    "source": [
-    "# Exercise 03: Models, data generation and Forward UQ\n",
+    "# Exercise 03: Forward models, data generation and forward UQ\n",
     "\n",
-    "In this notebook, we dive into models in CUQIpy, generating simulated noisy data and some forward uncertainty quantification.\n",
+    "In this notebook, we dive into forward models in CUQIpy. In brief, a forward model is defined as a mathematical model that describes the relationship between some input parameters (which are not directly observable) and some output data (which is observable).\n",
     "\n",
-    "Finally, we also show how to define new custom models in CUQIpy from either matrices or functions (methods in python).\n",
+    "In the context of inverse problems forward models are an important concept. What distinguishes forward models for inverse problems to any other kind of model of a physical system, is that they are *ill-posed*, which means existence, uniqueness and stability of inversion cannot be guaranteed. This deserves a separate discussion, which is out-of-scope for this notebook and we refer to [1] for the interested reader.\n",
```
Comment (jakobsj) [inline training/Exercise03_Forward_Models.ipynb:11]:
```suggestion
    "In the context of inverse problems forward models are an important concept. What distinguishes forward models for inverse problems to any other kind of model of a physical system, is that they are typically *ill-posed*, which means that existence, uniqueness and/or stability of inversion cannot be guaranteed. This deserves a separate discussion, which is out-of-scope for this notebook and we refer to [1] for the interested reader.\n",
```

File: training/Exercise03_Forward_Models.ipynb [hunk 2]
```diff
@@ -40,18 +47,15 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# 1. Access models from test problems<a class=\"anchor\" id=\"testproblems\"></a>\n",
-    "Models in CUQIpy are the link between the solution/parameter, say $\\mathbf{x}$, and the data, say $\\mathbf{y}$. In their simplest form they are simply a mapping $A: \\mathbf{x} \\mapsto \\mathbf{y}$.\n",
+    "# 1. Access forward models from test problems<a class=\"anchor\" id=\"testproblems\"></a>\n",
+    "Forward models in CUQIpy are the link between the parameter of interest (which we call solution), say $\\mathbf{x}$, and the observed data, say $\\mathbf{y}$. In their simplest form they are simply a mapping $A: \\mathbf{x} \\mapsto \\mathbf{y}$. In CUQIpy forward models are so commonly used, that we refer to them simply as models.\n",
```
Comment (jakobsj) [inline training/Exercise03_Forward_Models.ipynb:58]: Good point that we refer to forward models as models.

File: training/Exercise03_Forward_Models.ipynb [hunk 3]
```diff
@@ -651,13 +659,13 @@
     "\n",
     "One of the main tasks when working on numerical experiments for inverse problems is to generate synthetic data (potentially many realizations) to test and validate against. \n",
     "\n",
-    "In this section, we demonstrate one way this can be achieved by defining the so-called *data distribution*.\n",
+    "In this section, we demonstrate one way this can be achieved by defining the *data distribution*.\n",
     "\n",
-    "Let us return to the model from the deconvolution testproblem from earlier, and let us assume that the measurement data is affected by additive i.i.d. Gaussian noise. This leads to the inverse problem\n",
+    "Let us return to the model from the deconvolution testproblem from earlier, and let us assume that the measurement data is affected by additive i.i.d. Gaussian noise with standard deviation $0.05$. This leads to the Bayesian model for the inverse problem\n",
```
Comment (jakobsj) [inline training/Exercise03_Forward_Models.ipynb:664]:
```suggestion
    "Let us return to the forward model from the deconvolution testproblem from earlier, and let us assume that the measurement data is affected by additive i.i.d. Gaussian noise with standard deviation $0.05$. This leads to the Bayesian model for the inverse problem\n",
```

File: training/Exercise03_Forward_Models.ipynb [hunk 4]
```diff
@@ -902,7 +912,7 @@
    "source": [
     "To match the picture above, we select the column-major order by defining `order=\"F\"` (F comes from Fortran for historical reasons).\n",
     "\n",
-    "Furthermore, by default the `Image2D` will convert the input to an image before being evaluated by the model. In our case the matrix expected a vector, so we do not want this and thus set `visual_only=True` to not convert the vector to an image before evaluating the model."
+    "By default `Image2D` converts the input to an image before being evaluated by the model. In our case the matrix expects a vector, so we do not want this default behavior and thus set `visual_only=True` to not convert the vector to an image before model evaluation."
```
Comment (jakobsj) [inline training/Exercise03_Forward_Models.ipynb:915]: Clearer like this - still a bit of a tricky point, but let's go with this.

File: training/Exercise03_Forward_Models.ipynb [hunk 5]
```diff
@@ -943,7 +953,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "For the row and column sum data we can for the purpose of the example let that be a Discrete geometry:"
+    "For the row and column sum data we can for the purpose of the example let that be a Discrete (labelled) geometry:"
```
Comment (jakobsj) [inline training/Exercise03_Forward_Models.ipynb:956]:
```suggestion
    "The row and column sum data we can - for the purpose of the example - let be of the type Discrete (labelled) geometry:"
```

**Review by jakobsj — APPROVED** [review]
Helpful clarifications of explanations. Pending minor suggested corrections this looks good to me.

**Pull request merged**